statuscheck/statuscheck.go

```diff
@@ -56,10 +57,10 @@ const (
 
 // Plugin struct holds all plugin-related data.
 type Plugin struct {
-	HTTP *httpmux.Plugin
-
-	transport datasync.TransportAdapter // data transport adapter
-	access    sync.Mutex                // lock for the Plugin data
+	HTTP     *httpmux.Plugin
+	Adapters *[]datasync.TransportAdapter // Injected transport
```

Comment:
This is ackward. I can not imagine from the name & even from comment what is the intent.